### PR TITLE
[core] Update format to match main v2

### PR DESCRIPTION
**Pull request by oliviertassinari**
Build on top of #1930, one step further

### CODE REVIEW

File: docs/pages/demo/datepicker/CustomDay.example.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
+import * as React from 'react';
 import clsx from 'clsx';
-import React, { useState } from 'react';
```
Comment (dmtrKovalenko) [inline docs/pages/demo/datepicker/CustomDay.example.js]: I am still aware of changing extensions. Let's use `jsx` for now. 
Or please provide any arguments for changing it.

**Review by dmtrKovalenko — CHANGES_REQUESTED** [review]
Lets get back to `jsx` for extensions.

File: docs/pages/demo/datepicker/CustomDay.example.js (hunk 1 above)
Comment (oliviertassinari) [reply docs/pages/demo/datepicker/CustomDay.example.js]: Ok, will revert

File: docs/pages/demo/datepicker/BasicDatePicker.example.jsx [hunk 2]
```diff
@@ -1,18 +0,0 @@
-import React, { useState } from 'react';
-import TextField from '@material-ui/core/TextField';
-import { DatePicker } from '@material-ui/pickers';
-
-function BasicDatePicker() {
```
Comment (oliviertassinari) [inline docs/pages/demo/datepicker/BasicDatePicker.example.jsx:5]: Seem to be a dead demo

**Review by dmtrKovalenko — APPROVED** [review]

**Review by dmtrKovalenko — APPROVED** [review]

**Pull request merged**